File: backend/nodeBackend/src/routes/auth.route.ts
import express from 'express';
import * as auth from '../controllers/auth.controller';
import { verifyRefreshToken, verifyToken } from '../middleware/auth.middleware';

const router = express.Router();

router.post('/login', auth.login);
router.post('/logout', auth.logout);
router.get('/refresh', verifyRefreshToken, auth.refresh);
router.get('/temp', verifyToken, auth.temp);

export { router as authRouter };

File: backend/authService/newImplementation/routes/auth.route.go
package routes

import (
	auth "github.com/ArmaanKatyal/tweetbit/backend/authService/controllers"
	"github.com/ArmaanKatyal/tweetbit/backend/authService/middlewares"
	"github.com/gin-gonic/gin"
)

type AuthRoute struct {
	rg *gin.RouterGroup
}

func NewAuthRoute(router *gin.RouterGroup) *AuthRoute {
	return &AuthRoute{
		rg: router,
	}
}

func (a *AuthRoute) HandleRoutes() {
	a.login()
	a.logout()
	a.refresh()
	a.register()
}

func (a *AuthRoute) login() {
	a.rg.POST("/login", auth.Login)
}

func (a *AuthRoute) logout() {
	a.rg.POST("/logout", auth.Logout)
}

func (a *AuthRoute) refresh() {
	rg := a.rg.Group("/refresh", middlewares.VerifyToken())
	rg.GET("/", auth.Refresh)
}

func (a *AuthRoute) register() {
	a.rg.POST("/register", auth.Register)
}
